### PR TITLE
SW-2003 Use jOOQ JSONB for arbitrary JSON objects

### PR DESCRIPTION
**Pull request by sgrimm**
In preparation for changing the default behavior of payload deserialization so
that it treats empty strings as null values, change the way we declare payload
fields that hold arbitrary client-supplied JSON objects. We will want such objects
to be preserved as-is, skipping the "treat empty strings as null" transformation,
which would be applied to `Map<String, Any?>` objects.

We could define our own class to hold arbitrary JSON objects, but since the values
will need to turn into jOOQ `JSONB` objects anyway, may as well just use that
class directly. In the payload classes, though, we'll refer to it by an alias both
to make the intent clearer and to reduce coupling between the API definition and
the database representation.

Update the user preferences API to use this new setup. This is purely an internal
change; it doesn't cause any client-visible behavior differences.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #766** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/766" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/766?utm_source=stack-comment).

File: openapi.yaml- [hunk 1]
```diff
@@ -0,0 +1,6541 @@
+openapi: 3.0.1
```
Comment (karthikbtf) [inline openapi.yaml-]: openapi.yaml- got picked up

**Review by karthikbtf — APPROVED** [review]
looks good to me 👍 
an extraneous file got picked up by mistake I think

File: openapi.yaml- (hunk 1 above)
Comment (sgrimm) [reply openapi.yaml-]: Oops. Thanks.

**Review by karthikbtf — APPROVED** [review]

**Pull request merged**